File: python/calculate_mode.py
def calculate_mode(ls):
    dictionary = {}

    # Create a dictionary of each item in the ls and their occurrences
    for item in ls:
        if item in dictionary:
            dictionary[item]+=1
        else:
            dictionary[item] = 1

    # Sort the dictionary by key in decending order
    sorted_dictionary = {k: v for k, v in sorted(dictionary.items(), key=lambda item: item[1], reverse=True)}

    # Create a list of lists of each item and how many times they occur
    occurrence_ls = [[k, v] for k, v in sorted_dictionary.items()]
    
    # Set the max occurrence as the first item in the list
    occurrence = int(occurrence_ls[0][1])
    
    mode = []

    # Loop through the lists of lists. If an item occurs more than or equal to what
    # the current highest occurrence is, push that item to the mode list
    for i in range(0, len(occurrence_ls)):
        if(int(occurrence_ls[i][1]) >= occurrence):
            mode.append(occurrence_ls[i][0])
    
    return mode

File: js/calculateMode.js
exports.calculateMode = (arr) => {
    myObj = {};

    for(let i=0; i<arr.length; i++) {
        if(arr[i] in myObj) {
            myObj[arr[i]]+=1;
        } else {
            myObj[arr[i]] = 1;
        }
    }

    let sortable = [];
    for (let item in myObj) {
        sortable.push([item, myObj[item]]);
    }
    
    sortable.sort(function(a, b) {
        return b[1] - a[1];
    });

    let occurrence = +(sortable[0][1]);

    let mode = [];

    for(let i=0; i<sortable.length; i++) {
        if(+(sortable[i][1]) >= occurrence) {
            mode.push(+sortable[i][0]);
        }
    }

    return mode;
}